src/tests/stress/pedsim/Ped_util/Regions.enc

```diff
@@ -0,0 +1,462 @@
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+              i = i + 1
+              x = x + dx
+              y = ymin
+            end
+            for yindex <- [0..n - 2] do
+              (this.boxes)(i) = new Box((xmax, y + dy - 1), (x, y))
+              if yindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+              win = win && link((this.boxes)(i), (this.boxes)(i - n))
+              i = i + 1
+              y = y + dy
+            end
+            (this.boxes)(i) = new Box((xmax, ymax), (x, y))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n))
+            win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+        else
+            this.boxes(0) = new Box((xmax,ymax), (xmin,ymin))
+        end
+        for a <- agents do
+          for b <- this.boxes do
+            if get(b!add(a)) then
+                break
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def move(i : int) : unit
+    for id <- [0..|this.boxes| - 1] do
+      (this.futures)(id) = ((this.boxes)(id)) ! move()
+    end
+    for f <- this.futures do
+      get(f)
+    end
+    if i > 1 then
+      this!move(i - 1)
+    else
+      print("DONE")
+    end
+  end
+  -- used by ui, if a ui is used
+  def agents() : [(int, int)]
+    val temp = new [[(int, int)]](|this.boxes|)
+    for i <- [0..|this.boxes| - 1] do
+      temp(i) = get((this.boxes(i)) ! agents())
+    end
+    var a=flatten(temp)
+    a
+  end
+end
+active class Box
+  var xmax : int
+  var ymax : int
+  var xmin : int
+  var ymin : int
+  var up : Maybe[Box]
+  var down : Maybe[Box]
+  var left : Maybe[Box]
+  var right : Maybe[Box]
+  var top_right : Maybe[Box]
+  var top_left : Maybe[Box]
+  var bottom_left : Maybe[Box]
+  var bottom_right : Maybe[Box]
+  var matrix : Quad_tree
+  var agents : Maybe[Item]
+  var last : Maybe[Item]
+  var newcommers : Maybe[Item]
+  var ttl : int
+  var size : int
+  var moving : bool
+  def init(in_max : (int, int), in_min : (int, int)) : unit
+    this.agents=Nothing
+    this.last=Nothing
+    this.newcommers=Nothing
+    this.up=Nothing
+    this.down=Nothing
+    this.left=Nothing
+    this.right=Nothing
+    this.top_left=Nothing
+    this.bottom_right=Nothing
+    this.bottom_left=Nothing
+    this.matrix = new Quad_tree(in_max, in_min)
+    this.top_right=Nothing
+    match in_max with
+      case (x, y) =>
+        this.xmax = x
+        this.ymax = y
+      end
+    end
+    match in_min with
+      case (x, y) =>
+        this.xmin = x
+        this.ymin = y
+      end
+    end
+    this.ttl = 10001
```

Comment:
removed it, it's the number of time steps +1, used for detecting agents that otherwise might move twice per time step.